src/test/java/com/clevertap/stormdb/StormDBTest.java

```diff
@@ -753,4 +760,58 @@ void testInMemoryUpdate() throws IOException, StormDBException {
         });
         assertEquals(3, db.size());
     }
+
+    @Test
+    public void testKeysDeletion() throws IOException, StormDBException{
+        final int valueSize = 28;
+        StormDB db = getStormDBInstance(valueSize);
+        // insert some keys
+        int totalEntries = 115;
+        ByteBuffer value = ByteBuffer.allocate(valueSize);
+        for (int i = 1; i <= totalEntries; i++) {
+            value.clear();
+            value.putInt(i);
+            db.put(i, value.array());
+        }
+
+        int[] total = {0};
+
+        db.iterate((key, data, offset) -> {
+            ByteBuffer dataValue = ByteBuffer.wrap(data, offset, valueSize);
+            assertEquals(key,   dataValue.getInt());
+            total[0]++;
+        });
+
+        assertEquals(totalEntries, total[0]);
+        // now delete every even number
+        total[0] = 0;
+
+        // simple deletion
+        for (int i = 1;i <= totalEntries; i++) {
+            if (i%2 == 0) {
+                db.remove(i);
+            }
+            if (i % 3 == 0) {
+                value.clear();
+                value.putInt(i);
+                db.put(i, value.array());
+            }
+        }
+        db.iterate((key, data, offset) -> {
+            ByteBuffer dataValue = ByteBuffer.wrap(data, offset, valueSize);
+            assertEquals(key,   dataValue.getInt());
+            total[0]++;
+        });
+        assertEquals(77, total[0]);
+
+        // compaction based deletion
```

Comment:
Can we also add a case which opens the DB, closes it, and then reopens to test the persistence of the deleted keys?